File: src/Monitor.App/Services/StatusService.cs
using Monitor.App.Models;
using Monitor.App.Repositories;

namespace Monitor.App.Services;

public class StatusService
{
    private readonly TimerService _timer;
    private readonly RemoteMonitorService _remoteMonitor;
    private readonly IUsageRepository _usageRepo;
    private readonly IReservationRepository _reservationRepo;

    public StatusService(TimerService timer, RemoteMonitorService remoteMonitor,
        IUsageRepository usageRepo, IReservationRepository reservationRepo)
    {
        _timer = timer;
        _remoteMonitor = remoteMonitor;
        _usageRepo = usageRepo;
        _reservationRepo = reservationRepo;
    }

    public AppStatus GetFullStatus()
    {
        var timerState = _timer.GetState();
        var remoteStatus = _remoteMonitor.GetStatus();

        string computerStatus;
        if (remoteStatus.Status == "error") computerStatus = "error";
        else if (remoteStatus.Status == "unknown") computerStatus = "unknown";
        else if (remoteStatus.IsRemote) computerStatus = "remote_controlled";
        else if (timerState.IsRunning) computerStatus = "in_use";
        else computerStatus = "idle";

        var todayStr = DateTime.Now.ToString("yyyy-MM-dd");

        return new AppStatus
        {
            Timestamp = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"),
            ComputerStatus = computerStatus,
            LocalUse = new LocalUseInfo
            {
                CurrentUser = timerState.CurrentUser,
                StartTime = timerState.StartTime,
                ElapsedSeconds = timerState.ElapsedSeconds,
                ElapsedFormatted = timerState.ElapsedFormatted
            },
            RemoteControl = new RemoteControlInfo
            {
                IsRemote = remoteStatus.IsRemote,
                RemoteType = remoteStatus.RemoteType,
                StartTime = remoteStatus.StartTime,
                ElapsedSeconds = remoteStatus.ElapsedSeconds,
                ElapsedFormatted = remoteStatus.ElapsedFormatted,
                Status = remoteStatus.Status,
                Source = remoteStatus.Source,
                Confidence = remoteStatus.Confidence,
                OperatorName = remoteStatus.OperatorName,
                LastSeenAt = remoteStatus.LastSeenAt,
                ErrorMessage = remoteStatus.ErrorMessage,
                MatchedSignals = remoteStatus.MatchedSignals,
                Message = remoteStatus.Message
            },
            TodayRecords = _usageRepo.GetTodayRecords(),
            TodayReservations = _reservationRepo.GetTodayReservations(todayStr)
        };
    }

    public SidebarStatus GetSidebarStatus()
    {
        var remoteStatus = _remoteMonitor.GetStatus();
        return new SidebarStatus
        {
            IsTiming = _timer.IsRunning,
            CurrentUser = _timer.IsRunning ? _timer.CurrentUserName : null,
            IsRemote = remoteStatus.IsRemote,
            RemoteType = remoteStatus.RemoteType,
            Status = remoteStatus.Status,
            ErrorMessage = remoteStatus.ErrorMessage
        };
    }
}

File: legacy/python-monitor/app/domain/services/status_service.py
"""
Status service — unified state aggregation entry point.
Provides a single source of truth for both desktop sidebar and web API.
"""
from datetime import datetime, date


class StatusService:
    """
    Aggregates local timer state, remote control state, today's records,
    and today's reservations into a single AppStatus dict.

    This is THE unified entry point for both GUI status panel and Web API.
    """

    def __init__(self, timer_service, remote_monitor_service,
                 usage_repo, reservation_repo):
        self._timer = timer_service
        self._remote = remote_monitor_service
        self._usage_repo = usage_repo
        self._reservation_repo = reservation_repo

    def get_full_status(self) -> dict:
        """Get complete aggregated status for GUI and Web consumption."""
        timer_state = self._timer.get_state()
        remote_state = self._remote.get_status()

        # Determine computer status
        if remote_state.get("is_remote"):
            computer_status = "remote_controlled"
        elif timer_state.get("is_running"):
            computer_status = "in_use"
        else:
            computer_status = "idle"

        today_records = self._usage_repo.get_today_records()
        today_str = date.today().strftime("%Y-%m-%d")
        today_reservations = self._reservation_repo.get_today_reservations(today_str)

        return {
            "timestamp": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
            "computer_status": computer_status,
            "local_use": {
                "current_user": timer_state.get("current_user"),
                "start_time": timer_state.get("start_time"),
                "elapsed_seconds": timer_state.get("elapsed_seconds", 0),
                "elapsed_formatted": timer_state.get("elapsed_formatted", "00:00:00"),
            },
            "remote_control": remote_state,
            "today_records": today_records,
            "today_reservations": today_reservations,
        }

    def get_sidebar_status(self) -> dict:
        """Get lightweight status for sidebar panel."""
        timer_state = self._timer.get_state()
        remote_state = self._remote.get_status()
        return {
            "is_timing": timer_state.get("is_running", False),
            "current_user": timer_state.get("current_user"),
            "is_remote": remote_state.get("is_remote", False),
            "remote_type": remote_state.get("remote_type"),
        }
